pkg/resource/resource.go

```diff
@@ -388,131 +388,172 @@ func (c *ResourceClient) CreateResourceStack(inventoryFile string, resourceConfi
 // DeleteResourceStack deletes all the resources in the resource inventory.
 func (c *ResourceClient) DeleteResourceStack(inventoryFile string) error {
 
-	resourceInv, err := ReadInventory(inventoryFile)
+	inventory, err := ReadInventory(inventoryFile)
 	if err != nil {
 		return err
 	}
 
-	c.AWSConfig.Region = resourceInv.Region
+	c.AWSConfig.Region = inventory.Region
 
 	// OIDC Provider
-	if err := c.DeleteOIDCProvider(resourceInv.OIDCProviderARN); err != nil {
+	if err := c.DeleteOIDCProvider(inventory.OIDCProviderARN); err != nil {
+		return err
+	}
+	err = WriteInventory(inventoryFile, inventory)
```

Comment:
For the sake of consistency, let's use this single-line syntax for catching errors from funcs that only return error...

```suggestion
	if err := WriteInventory(inventoryFile, inventory); err != nil {
```